tests/templates/test_subroutines.py

```diff
@@ -472,3 +474,163 @@ def test_correct_gates_two_wires(self):
+                       [97,  qml.Hadamard, [9], []]     , [98, qml.Hadamard, [10], []],
+                       [104, qml.RZ, [10], [-np.pi/24]] ,
+                       [110, qml.Hadamard, [4], []]     , [111, qml.RX,       [7] , [np.pi/2]],
+                       [112, qml.Hadamard, [9], []]     , [113, qml.Hadamard, [10], []]]
+                   ),
+        ([5,6,10,12], [[102, qml.RX, [5],  [-np.pi/2]]  , [103, qml.RX,       [6],  [-np.pi/2]],
+                       [104, qml.RX, [10], [-np.pi/2]]  , [105, qml.Hadamard, [12], []],
+                       [110, qml.RZ, [12], [-np.pi/24]] ,
+                       [115, qml.RX, [5],  [np.pi/2]]   , [116, qml.RX,       [6],  [np.pi/2]],
+                       [117, qml.RX, [10], [np.pi/2]]   , [118, qml.Hadamard, [12], []]]
+                   ),
+        ([3,6,17,19], [[147, qml.RX,       [3],  [-np.pi/2]], [148, qml.RX, [6],  [-np.pi/2]],
+                       [149, qml.Hadamard, [17], []]        , [150, qml.RX, [19], [-np.pi/2]],
+                       [157, qml.RZ, [19], [-np.pi/24]]     ,
+                       [164, qml.RX, [3],  [np.pi/2]]       , [165, qml.RX, [6],  [np.pi/2]],
+                       [166, qml.Hadamard, [17], []]        , [167, qml.RX, [19], [np.pi/2]]]
+                   ),
+        ([6,7,8,9]  , [[4, qml.CNOT, [6, 7], []], [5,  qml.CNOT, [7, 8], []], 
+                       [6, qml.CNOT, [8, 9], []], [8,  qml.CNOT, [8, 9], []],
+                       [9, qml.CNOT, [7, 8], []], [10, qml.CNOT, [6, 7], []]]
+                   ),
+        ([4,7,8,13] , [[58,  qml.CNOT, [4, 5],   []], [59, qml.CNOT, [5, 6],   []], 
+                       [60,  qml.CNOT, [6, 7],   []], [61, qml.CNOT, [7, 8],   []],
+                       [62,  qml.CNOT, [8, 9],   []], [63, qml.CNOT, [9, 10],  []],
+                       [64,  qml.CNOT, [10, 11], []], [65, qml.CNOT, [11, 12], []],
+                       [66,  qml.CNOT, [12,13],  []],
+                       [122, qml.CNOT, [12, 13], []], [123, qml.CNOT, [11, 12], []], 
+                       [124, qml.CNOT, [10, 11], []], [125, qml.CNOT, [9, 10],  []],
+                       [126, qml.CNOT, [8, 9],   []], [127, qml.CNOT, [7, 8],   []],
+                       [128, qml.CNOT, [6, 7],   []], [129, qml.CNOT, [5, 6],   []],
+                       [130, qml.CNOT, [4,5],    []]]
+                   ),
+        ]
+    )
+    def test_double_ex_unitary_operations(self, pphh, ref_gates):
+        """Test the correctness of the DoubleExcitationUnitary template including the gate count
+        and order, the wires each operation acts on and the correct use of parameters 
+        in the circuit."""
+
+        sqg = 72
+        cnots = 16*(pphh[1]-pphh[0] + pphh[3]-pphh[2] + 1)
+        weight = np.pi/3
+        with qml.utils.OperationRecorder() as rec:
+            DoubleExcitationUnitary(weight, wires=pphh)
+
+        idx = ref_gates[0][0]
```

Comment:
I was curious what the idx represents?

Reply:
`idx` above is the gate index as they appear in `rec.queue`. The test checks for a given index, the gate, the wire(s) it acts on and its argument.